### PR TITLE
Fix a few typos and styling issues in RenderPassDescriptor validation

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
It also fixed a couple styling issues, and typos, etc.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on Sep 20, 2021, 6:20 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2FRichard-Yunchao%2Fgpuweb%2Fce356baee8fc93e8d91b0ed14ce59f0cb0955015%2Fspec%2Findex.bs&force=1&md-warning=not%20ready)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%232121.)._
</details>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -7142,22 +7142,21 @@ dictionary GPURenderPassDepthStencilAttachment {
...
-        - |this|.{{GPURenderPassDepthStencilAttachment/depthLoadValue}} must be {{GPULoadOp/"load"}}.
-        - |this|.{{GPURenderPassDepthStencilAttachment/depthStoreOp}} must be {{GPUStoreOp/"store"}}.
-    - If |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}} is `true`,
-        |this|.{{GPURenderPassDepthStencilAttachment/stencilLoadValue}} must be
-        {{GPULoadOp/"load"}} and |this|.{{GPURenderPassDepthStencilAttachment/stencilStoreOp}}
-        must be {{GPUStoreOp/"store"}}.
+        1. |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} must be equal to |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}}
+    1. If |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} is `true`:
+        1. |this|.{{GPURenderPassDepthStencilAttachment/depthLoadValue}} must be {{GPULoadOp/"load"}}.
+        1. |this|.{{GPURenderPassDepthStencilAttachment/depthStoreOp}} must be {{GPUStoreOp/"store"}}.
```
Comment (Richard-Yunchao) [inline spec/index.bs:7156]: Do we need to add "|this|.{{GPURenderPassDepthStencilAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}} must contain depth aspect", and a similar one for stencil?

Comment (kvark) [reply spec/index.bs:7156]: good idea!

File: spec/index.bs [hunk 2]
```diff
@@ -7142,22 +7142,21 @@ dictionary GPURenderPassDepthStencilAttachment {
     Given a {{GPURenderPassDepthStencilAttachment}} |this| the following validation
     rules apply:
 
-    - |this|.{{GPURenderPassDepthStencilAttachment/view}} must have a [=depth or stencil renderable
+    1. |this|.{{GPURenderPassDepthStencilAttachment/view}} must have a [=depth or stencil renderable
```
Comment (kvark) [inline spec/index.bs:7145]: why are we changing the "-" into "1" here? The list is not ordered.

**Review by kvark — CHANGES_REQUESTED** [review]

File: spec/index.bs (hunk 1 above)
Comment (Richard-Yunchao) [reply spec/index.bs]: Done.

File: spec/index.bs (hunk 2 above)
Comment (Richard-Yunchao) [reply spec/index.bs]: I noticed that the validation rules for RenderPassDesc and ColorAttachment are in ordered list. But it is not necessary to be ordered, and the majority of validation rules in the spec are not in ordered list (Although they are not always consistent). So I recovered this to what it is. My bad...

**Review by kvark — APPROVED** [review]

File: spec/index.bs [hunk 3]
```diff
@@ -6979,7 +6979,7 @@ dictionary GPURenderPassDescriptor : GPUObjectDescriptorBase {
 
     1. Each {{GPURenderPassColorAttachment/view}} in |this|.{{GPURenderPassDescriptor/colorAttachments}}
         and |this|.{{GPURenderPassDescriptor/depthStencilAttachment}}.{{GPURenderPassDepthStencilAttachment/view}},
```
Comment (kainino0x) [inline spec/index.bs:6981]: Slightly better:

```suggestion
    1. All {{GPURenderPassColorAttachment/view}}s in |this|.{{GPURenderPassDescriptor/colorAttachments}},
        and |this|.{{GPURenderPassDescriptor/depthStencilAttachment}}.{{GPURenderPassDepthStencilAttachment/view}}
```

File: spec/index.bs [hunk 4]
```diff
@@ -6979,7 +6979,7 @@ dictionary GPURenderPassDescriptor : GPUObjectDescriptorBase {
...
-        if present, must have all have the same {{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/sampleCount}}.
+        if present, must all have the same {{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/sampleCount}}.
```
Comment (kainino0x) [inline spec/index.bs:6982]:
```suggestion
        if present, must have equal {{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/sampleCount}}s.
```

File: spec/index.bs [hunk 5]
```diff
@@ -7152,12 +7152,13 @@ dictionary GPURenderPassDepthStencilAttachment {
         contains both depth and stencil aspects (See [[#depth-formats|depth-stencil formats]]):
         - |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} must be equal to |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}}
     - If |this|.{{GPURenderPassDepthStencilAttachment/depthReadOnly}} is `true`:
+        - |this|.{{GPURenderPassDepthStencilAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}} must contain depth aspect.
```
Comment (kainino0x) [inline spec/index.bs:7155]:
```suggestion
        - |this|.{{GPURenderPassDepthStencilAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}} must contain a depth aspect.
```

File: spec/index.bs [hunk 6]
```diff
@@ -7152,12 +7152,13 @@ dictionary GPURenderPassDepthStencilAttachment {
...
         - |this|.{{GPURenderPassDepthStencilAttachment/depthLoadValue}} must be {{GPULoadOp/"load"}}.
         - |this|.{{GPURenderPassDepthStencilAttachment/depthStoreOp}} must be {{GPUStoreOp/"store"}}.
-    - If |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}} is `true`,
-        |this|.{{GPURenderPassDepthStencilAttachment/stencilLoadValue}} must be
-        {{GPULoadOp/"load"}} and |this|.{{GPURenderPassDepthStencilAttachment/stencilStoreOp}}
-        must be {{GPUStoreOp/"store"}}.
+    - If |this|.{{GPURenderPassDepthStencilAttachment/stencilReadOnly}} is `true`:
+        - |this|.{{GPURenderPassDepthStencilAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}} must contain stencil aspect.
```
Comment (kainino0x) [inline spec/index.bs:7159]:
```suggestion
        - |this|.{{GPURenderPassDepthStencilAttachment/view}}.{{GPUTextureView/[[descriptor]]}}.{{GPUTextureViewDescriptor/format}} must contain a stencil aspect.
```

File: spec/index.bs (hunk 1 above)
Comment (kainino0x) [reply spec/index.bs]: If I'm reading correctly this is a new validation rule, right? @kvark I don't think this really makes sense. You could just as easily specify `depthReadOnly: false` explicitly, why should that be allowed but `depthReadOnly: true` be disallowed?

I don't think this needs a validation rule unless it's adding something I don't understand.

Comment (Richard-Yunchao) [reply spec/index.bs]: If `depthReadOnly` is `false`, maybe some combinations with `depthStoreOp` and `depthLoadValue` also require format contain depth aspect? I mean, the textureView attached should be explicitly required to contain depth aspect when it is definitely going to operate on depth aspect for some combinations via setting these values. The question is: what are the combinations? If `depthReadOnly` is explicitly set to true, I am not 100% sure, but my understanding is that we are going to read data from depth aspect from depth/stencil attachment. It is similar for stencil aspect. WDYT, Kai?

Comment (kainino0x) [reply spec/index.bs]: `depthReadOnly: true` only prevents writing, it has no bearing on whether depth is read or not. (`setPipeline` [validates](https://gpuweb.github.io/gpuweb/#dom-gpurenderencoderbase-setpipeline) the pipeline against the `depthReadOnly`/`stencilReadOnly` of the pass.) It's the "this.[[layout]] equals pipelineTargetsLayout" rule that validates a pipeline's (color and depth) attachments against the render pass. That requires the exact formats to match. And #2079 requires the GPUTextureView to reference all aspects.

I don't think createRenderPipeline has any validation to ensure depth-writing flags aren't used with depth-less stencil formats, but I think it's implicit in the pipeline creation that it won't do anything?

Comment (Richard-Yunchao) [reply spec/index.bs]: OK. It makes sense. Thanks for the details. 

I removed that incorrect validation rules. Now the PR just correct some typos and style stuff. Please help to merge it if it is OK, @kainino0x .

Comment (kainino0x) [reply spec/index.bs]: LGTM. @kvark please also review this thread in case there's some edit you still want to make to this validation logic.

**Review by kainino0x — APPROVED** [review]

**Pull request merged**